PACKAGING-GUIDE.md

```diff
@@ -0,0 +1,100 @@
+
+The format of the catalog is JSON. It is possible for users to have a custom version of
+the catalog. In order to do so, copy the file located at https://github.com/projectatomic/ai-studio/blob/main/packages/backend/src/ai.json
+to $HOME/podman-desktop/ai-studio/catalog.json and AI Studio will use it instead of the embedded one.
+Any change done to this file will also be automatically loaded by AI Studio.
+
+### Format of the catalog file
+
+The catalog file has three main elements: categories, models and recipes. Each of these elements is
+represented in the JSON file as an array.
+
+#### Categories
+
+This is the top level construct of the catalog UI. Recipes are grouped into categories. A category
+represents the kind of AI application. Although the list of categories provided by default by
+AI Studio represents the AI landscape, it is possible to add new categories.
+
+A category has three main attributes: an id (which should be unique among categories), a description
+and a name. The category id attribute will then be used to attach a recipe to one or several categories.
+
+#### Models
+
+The catalog also list the models that may be associated to recipes. A model is also a first class
+citizen in AI Studio as they will be listed in the Models page and can be tested through the playground.
+
+A model has the following attributes:
+- ```id```: a unique identifier for the model
+- ```name```: the model name
+- ```description```: a detailed description about the model
+- ```hw```: the hardware where the model is compatible. Possible values are CPU and GPU
+- ```registry```: the model registry where the model is stored
+- ```popularity```: an integer field giving the rating of the model. Can be thought as the number of stars
+- ```license```: the license under which the model is available
+- ```url```: the URL used to download the model
+
+#### Recipes
+
+A recipe is a sample AI application that is packaged as one or several containers. It is built by AI Studio when the user chooses to download and run it on their workstation. It is provided as
+source code and AI Studio will make sure the container images are built prior to launching the containers.
+
+A recipe has the following attributes:
+- ```id```: a unique identifier to the recipe
+- ```name```: the recipe name
+- ```description```: a detailed description about the recipe
+- ```repository```: the URL where the recipe code can be retrieved
+- ```categories```: an array of category id to be associated by this recipe
+- ```config```: an optional path of the configuration file within the repository. If not provided, the file is assumed to be located at the root the repository and called ai-studio.yaml
+- ```readme```: a markdown description of the recipe
+- ```models```: an array of model id to be associated with this recipe
+
+#### Recipe configuration file
+
+The configuration file is called ```ai-studio.yaml``` and follows the following syntax.
```

Comment:
Where do we need to place this file?

Reply:
See the config attribute of a recipe. Maybe I should repeat or link to the definition

Reply:
Yes, I think it would be nice to have the context, for example saying: 'The configuration file for a recipe, as defined in the `config` field ...'